cpp/include/cudf/scalar/scalar_device_view.cuh

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/strings/string_view.cuh>
+#include <cudf/types.hpp>
+
+namespace cudf {
+
+namespace detail {
+
+/**---------------------------------------------------------------------------*
+ * @brief An immutable, non-owning view of scalar from device that is trivially 
+ * copyable and usable in CUDA device code.
+ *---------------------------------------------------------------------------**/
+class alignas(16) scalar_device_view_base {
+ public:
+  scalar_device_view_base() = default;
+  ~scalar_device_view_base() = default;
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns the element type
+   *---------------------------------------------------------------------------**/
+  __host__ __device__ data_type type() const noexcept { return _type; }
+
+  /**---------------------------------------------------------------------------*
+   * @brief Returns whether the scalar holds a valid value (i.e., not null).
+   *
+   * @return true The element is valid
+   * @return false The element is null
+   *---------------------------------------------------------------------------**/
+  __device__ bool is_valid() const noexcept {
```

Comment:
Do we need `noexcept` on device-only functions?

Reply:
Not really, no. No harm in it either.